spaceline-all-the-icons-segments.el

```diff
@@ -465,9 +465,9 @@ doesn't inherit all properties of a face."
     (if (not (and spaceline-all-the-icons-highlight-file-name
                   show-path?))
         (add-to-list 'file-face :inherit t)
-      (plist-put file-face :background (face-background default-face))
-      (plist-put file-face :foreground (or spaceline-all-the-icons-file-name-highlight
-                                           (face-background highlight-face))))
+      (setq file-face (append `(:background ,(face-background default-face)) file-face))
+      (setq file-face (append `(:foreground (or spaceline-all-the-icons-file-name-highlight
```

Comment:
Last comma, needs to be before the `(or` function call rather than just the `(face-background` call, _i.e._

```el
,(or spaceline-all-the-icons-file-name-highlight
     (face-background highlight-face))
```

Then this is great to go! :)